distributed/worker.py

```diff
@@ -298,7 +298,11 @@ def _close(self, report=True, timeout=10):
             v.stop()
         self.rpc.close()
         self.status = 'closed'
-        self.stop()
+        with ignoring(gen.TimeoutError):
+            yield gen.with_timeout(timedelta(seconds=1),
+                    [close(stream) for stream in self._listen_streams])
+        for stream in self._listen_streams:
+            stream.close()
```

Comment:
It may be that closing the stream is confusing another coroutine which is using the stream at the same time. My question is: why did you want to add this?